apps/condo/domains/common/utils/filters.utils.ts

```diff
@@ -0,0 +1,128 @@
+    props?: InputProps
+} | {
+    type: ComponentType.CheckboxGroup
+    options: { value: string, label: string }[]
+    loading: boolean
+    props?: CheckboxGroupProps
+} | {
+    type: ComponentType.Select
+    options: { value: string, label: string }[]
+    props?: SelectProps<string>
+} | {
+    type: ComponentType.Date
+    props?: DatePickerProps
+} | {
+    type: ComponentType.DateRange
+    props?: RangePickerSharedProps<Dayjs>
+} | {
+    type: ComponentType.Custom
+    getComponentFilterDropdown?: ({ setSelectedKeys, selectedKeys, confirm, clearFilters }: {
+        setSelectedKeys: any, selectedKeys: any, confirm: any, clearFilters: any
+    }) => JSX.Element
+    modalFilterComponent?: React.ReactElement
+})
+
+export type FiltersMeta<F> = QueryMeta<F> & {
+    component?: FilterComponentInfo,
+}
+
+export const getQueryToValueProcessorByType = (type: ComponentType) => {
+    switch (type) {
+        case ComponentType.Date:
+            return value => dayjs(value)
+        case ComponentType.DateRange:
+            return value => value.map(date => dayjs(date))
+        default: return
+    }
+}
+
+export function getFilterDropdownByKey <T> (filterMetas: Array<FiltersMeta<T>>, key: string) {
+    const filterMeta = filterMetas.find(filterMeta => filterMeta.keyword === key)
+    const component = get(filterMeta, 'component')
+    const type = get(component, 'type')
+    const props = get(component, 'props')
+    const columnFilterComponentWrapperStyles = get(component, 'columnFilterComponentWrapper')
+
+    switch (type) {
+        case ComponentType.Input: {
+            const placeholder = get(props, 'placeholder')
+            return getTextFilterDropdown(placeholder, columnFilterComponentWrapperStyles)
+        }
+
+        case ComponentType.Date:
+            return getDateFilterDropdown(columnFilterComponentWrapperStyles)
+
+        case ComponentType.CheckboxGroup: {
+            const options = get(component, 'options')
+            const loading = get(component, 'loading')
+            return getOptionFilterDropdown(options, loading, columnFilterComponentWrapperStyles)
+        }
+
+        case ComponentType.DateRange:
+            return getDateRangeFilterDropdown(columnFilterComponentWrapperStyles)
+
+        case ComponentType.Select: {
+            const options = get(component, 'options')
+            const loading = get(component, 'loading')
+            const mode = get(component, ['props', 'mode'])
+            return getSelectFilterDropdown(options, loading, mode, columnFilterComponentWrapperStyles)
+        }
+
+        case ComponentType.GQLSelect: {
+            const search = get(props, 'search')
+            const mode = get(props, 'mode')
+            return getGQLSelectFilterDropdown(search, mode, columnFilterComponentWrapperStyles)
+        }
+
+        case ComponentType.Custom: {
+            return get(component, 'getComponentFilterDropdown')
+        }
+
+        default: return
```

Comment:
May be better to throw an error like `Unknown component type`?

Reply:
It seems to me it is better to return void, because in this case we simply will not have a component for filtering (because we did not find the right one by type). If you throw out an error, the client part will fall in runtime
<img width="633" alt="изображение" src="https://user-images.githubusercontent.com/52532264/134321193-ae4936e4-9752-42ac-ad3a-f35fd1299585.png">